internal/snips/file.go

```diff
@@ -58,3 +61,33 @@ func (f *File) Visibility() string {
 
 	return "public"
 }
+
+var decoder, _ = zstd.NewReader(nil)
+
+func (f *File) GetContent() ([]byte, error) {
+	// check if it has been compressed
+	if !f.isCompressed() {
+		return f.content, nil
+	}
+
+	defer decoder.Close()
+	decodedBytes, err := decoder.DecodeAll(f.content, nil)
+
+	return decodedBytes, err
+}
+
+var encoder, _ = zstd.NewWriter(nil)
+
+func (f *File) SetContent(in []byte) {
+	f.content = encoder.EncodeAll(in, nil)
+}
+
+func (f *File) isCompressed() bool {
+	magic := []byte{0x28, 0xb5, 0x2F, 0xFD}
+	for i := 0; i < 4; i++ {
+		if f.content[i] != magic[i] {
+			return false
+		}
+	}
+	return true
```

Comment:
```suggestion
	// check if first 4 bytes are ZSTD magic number
	// https://github.com/facebook/zstd/blob/dev/doc/zstd_compression_format.md#zstandard-frames
	return len(data) > 4 && binary.BigEndian.Uint32(data) == 0x28B52FFD
```

got a nice one-liner for this (and a comment!)

Reply:
Fancy 🕺